server-nodejs/extensions/jwt.js

```diff
@@ -96,24 +82,40 @@ module.exports = function (app) {
 
     app.use(cookieParser());
 
-    app.use(function (req, res, next){
-        var token = fetch(req.headers) || req.cookies.token;
-        if (null === token) {
+    app.use( function (req, res, next) {
+        let token = fetch(req.headers) || req.cookies.token;
+        let obj = jwt.decode(token);
+        if ((undefined === token) || (null === token) || (null === obj)) {
             req.user = {};
             return next();
         }
-        jwt.verify(token, conf.secret, function (err, decode) {
-            if (err) {
-                req.user = {};
-                return next();
-            }
-            db.read([decode._id], function (err, user) {
-                req.user = user[0];
-                next();
+        db.read([obj['_id']], function (err, user) {
+            let password = user[0]['password'];
+            let salt = conf.secret + password;
+            jwt.verify(token, salt, function (err, decode) {
```

Comment:
one-time-use variables could  be avoided
 ```suggestion
            jwt.verify(token, conf.secret + user[0]['password'], function (err, decode) {
```